flang/lib/Lower/OpenACC.cpp

```diff
@@ -15,38 +15,57 @@
 #include "flang/Lower/FIRBuilder.h"
 #include "flang/Lower/PFTBuilder.h"
 #include "flang/Parser/parse-tree.h"
+#include "flang/Semantics/tools.h"
+#include "mlir/Dialect/OpenACC/OpenACC.h"
 #include "llvm/Frontend/OpenACC/ACC.h.inc"
 
 #define TODO() llvm_unreachable("not yet implemented")
 
-void Fortran::lower::genOpenACCConstruct(
+void Fortran::lower::genACC(
     Fortran::lower::AbstractConverter &absConv,
     Fortran::lower::pft::Evaluation &eval,
-    const Fortran::parser::OpenACCConstruct &accConstruct) {
-
-  std::visit(
-      common::visitors{
-          [&](const Fortran::parser::OpenACCBlockConstruct &blockConstruct) {
-            TODO();
-          },
-          [&](const Fortran::parser::OpenACCCombinedConstruct
-                  &combinedConstruct) { TODO(); },
-          [&](const Fortran::parser::OpenACCLoopConstruct &loopConstruct) {
-            TODO();
-          },
-          [&](const Fortran::parser::OpenACCStandaloneConstruct
-                  &standaloneConstruct) { TODO(); },
-          [&](const Fortran::parser::OpenACCRoutineConstruct
-                  &routineConstruct) { TODO(); },
-          [&](const Fortran::parser::OpenACCCacheConstruct &cacheConstruct) {
-            TODO();
-          },
-          [&](const Fortran::parser::OpenACCWaitConstruct &waitConstruct) {
-            TODO();
-          },
-          [&](const Fortran::parser::OpenACCAtomicConstruct &atomicConstruct) {
-            TODO();
-          },
-      },
-      accConstruct.u);
+    const Fortran::parser::OpenACCLoopConstruct &loopConstruct) {
+
+  const auto &beginLoopDirective =
+      std::get<Fortran::parser::AccBeginLoopDirective>(loopConstruct.t);
+  const auto &loopDirective =
+      std::get<Fortran::parser::AccLoopDirective>(beginLoopDirective.t);
+
+  if (loopDirective.v == llvm::acc::ACCD_loop) {
+    auto &firOpBuilder = absConv.getFirOpBuilder();
+    auto currentLocation = absConv.getCurrentLocation();
+    auto insertPt = firOpBuilder.saveInsertionPoint();
+    llvm::ArrayRef<mlir::Type> argTy;
+    mlir::ValueRange range;
+    llvm::SmallVector<int32_t, 2> operandSegmentSizes(2 /*Size=*/,
+                                                      0 /*Value=*/);
```

Comment:
The comments should come before the constants (i.e. `/*Size=*/ 2`) for consistency.